draft-ietf-jose-hpke-encrypt.md

```diff
@@ -376,127 +799,183 @@ The guidance on randomness in {{RFC4086}} applies.
 
 ## Key Management
 
-A single KEM key MUST NOT be used with multiple KEM algorithms.
-Each key and its associated algorithm suite, comprising the KEM, KDF, and AEAD,
-should be managed independently.
+A single key MUST NOT be used with multiple KEM algorithms.
+Each key and its associated HPKE algorithm suite, comprising the KEM, KDF, and AEAD,
+SHOULD be managed independently.
 This separation prevents unintended interactions or vulnerabilities between algorithms,
 ensuring the integrity and security guarantees of each algorithm are preserved.
-Additionally, the same key should not be used for both key encryption and integrated encryption, as it may introduce security risks.
+Additionally, the same key SHOULD NOT be used for both
```

Comment:
we will have to provide a justification for the exception, I prefer "MUST" instead.

Reply:
That would be another RFC 6919 "MUST (BUT WE KNOW YOU WON'T)", because in practice, very few applications are going to do that.

And in practice, one can not even rely on explicit checks working. So HPKE in JWE must be designed to resist any such attacks at cryptographic level. Fortunately, that is the case.

Reply:
Implementations can ignore it, but that is also true for similar normative usage requirements present in HPKE, such as not reusing ikm across KEMs or not reusing randomness in KEM operations. These requirements cannot be enforced by the protocol either, yet HPKE still states them normatively because they are essential for security. Our use of “MUST” fits the same model.